### PR TITLE
Moved and simplified bv::utils::intersect.

### PR DESCRIPTION
**Pull request by aniemetz**
Tested against the recursive implementation with a temporary assertion on regression tests
with --bv-eq-slicer=auto.

### CODE REVIEW

File: src/theory/bv/slicer.cpp [hunk 1]
```diff
@@ -20,13 +20,35 @@
...
+               const std::vector<TermId>& v2,
+               std::vector<TermId>& intersection)
+{
+  for (const TermId id1 : v1)
+  {
+    for (const TermId id2 : v2)
+    {
+      if (v2[id2] == v1[id1])
+      {
+        intersection.push_back(id1);
```
Comment (mpreiner) [inline src/theory/bv/slicer.cpp]: This should be `intersection.push_back(v1[id1])` right?

File: src/theory/bv/slicer.cpp [hunk 2]
```diff
@@ -20,13 +20,35 @@
...
+
+void intersect(const std::vector<TermId>& v1,
+               const std::vector<TermId>& v2,
+               std::vector<TermId>& intersection)
+{
+  for (const TermId id1 : v1)
+  {
+    for (const TermId id2 : v2)
+    {
+      if (v2[id2] == v1[id1])
```
Comment (mpreiner) [inline src/theory/bv/slicer.cpp]: Shouldn't this be `id2 == id1`?

Comment (aniemetz) [reply src/theory/bv/slicer.cpp]: Good catch. Interestingly, when I tested the new against the old implementation, no discrepancy was encountered.

Comment (aniemetz) [reply src/theory/bv/slicer.cpp]: Maybe I forgot to enable the slicer, that would be the only explanation for that I can think of...

File: src/theory/bv/slicer.cpp [hunk 3]
```diff
@@ -381,12 +403,12 @@ void UnionFind::alignSlicings(const ExtractTerm& term1, const ExtractTerm& term2
 
   Assert (nf1.base.getBitwidth() == nf2.base.getBitwidth());
   
-  // first check if the two have any common slices 
+  // first check if the two have any common slices
   std::vector<TermId> intersection; 
-  utils::intersect(nf1.decomp, nf2.decomp, intersection); 
-  for (unsigned i = 0; i < intersection.size(); ++i) {
-    // handle common slice may change the normal form
```
Comment (mpreiner) [inline src/theory/bv/slicer.cpp]: Is there a reason why you removed the comment?

Comment (aniemetz) [reply src/theory/bv/slicer.cpp]: Oh. No. Accident. Will readd.

**Pull request merged**